pkg/virt-handler/device-manager/mediated_devices_types.go

```diff
@@ -72,6 +80,7 @@ func (m *MDEVTypesManager) updateMDEVTypesConfiguration(desiredTypesList []strin
 		}
 		// store the configured list of types
 		m.configuredMdevTypes = desiredTypesBytes
+		m.discoveredMdevTypesCount = len(files)
```

Comment:
I am not entirely sure if just taking the count is good enough. Would it make sense to just always create the updated list and then just compare if something changed?

Reply:
Yes, I went back and forth on this.
The issue is that `discoverConfigurableMDEVTypes` may be dependent on the results of `removeUndesiredMDEVs`
But,  `removeUndesiredMDEVs` should run only if there was a change in `desiredTypesBytes`
That's why I thought that counting the types would be an easy workaround, but you're right. It's not enough.

On the other hand, I think just keeping a list of PCI cards that provide mdevs, should be enough.
I'll post an update.

While looking into this again, I've discovered that we're not keeping track of cards that are already configured. That way updateMDEVTypesConfiguration may not take it into account correctly.
I'll update this as well now.